variables.tf

```
@@ -147,6 +147,12 @@ variable "use_amazon_linux2" {
   description = "Use Amazon Linux 2 instead of Amazon Linux"
 }
 
+variable "use_amazon_linux2023" {
+  description = "Use Amazon Linux 2023 AMI"
+  type        = bool
+  default     = true
```

By making this false, you could avoid a major version increase. Alternatively, you could go ahead with a major version increase and drop support for AL1, making the code simpler.
```suggestion
  default     = false
```